mixer/pkg/runtime/handler/logger.go

```diff
@@ -84,3 +65,28 @@ func (l logger) Errorf(format string, args ...interface{}) error {
 	l.s.Error(s)
 	return errors.New(s)
 }
+
+// Debugf from adapter.Logger.
+func (l logger) Debugf(format string, args ...interface{}) {
+	l.s.Debugf(format, args...)
```

Comment:
I think it might be worth doing `if debugEnabled` here as well, unless `l.s.Debugf` is already doing it. Most of the time debugEnabled is false, so doing a blind check, in case the caller has forgotten, will save overhead of the underlying `sprintf`

Reply:
It's done inside, before the sprintf kicks in. So I think we're OK.